### PR TITLE
[Custom Environments] Add new flag `--target` to the `redeploy` command

### PR DESCRIPTION
**Pull request by paulogdm**
# Pull Request

In this PR, we are adding a new command option to the `redeploy` command. With the `--target` flag, the CLI should be capable of:

- Redeploying a deployment to another environment. For example, a `production` deployment can now be deployed to `preview`.
- Redeploying a deployment to custom environments. If a deployment exists, we should be able to redeploy it in any other environment.

### CODE REVIEW

File: packages/cli/src/commands/redeploy/command.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ export const redeployCommand = {
       type: Boolean,
       deprecated: false,
     },
+    {
+      name: 'environment',
```
Comment (trek) [inline packages/cli/src/commands/redeploy/command.ts]: Within the deploy/build parts of the CLI this is referred to as `target`. Not sure _why_. But this should probably match here so 

```
vercel deploy --target=qa
vercel redeploy dpl_123 --target=qa
```

have parity.

**Review by trek — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/redeploy/command.ts (hunk 1 above)
Comment (paulogdm) [reply packages/cli/src/commands/redeploy/command.ts]: Thanks! Changed. But 🤢

File: packages/cli/src/commands/redeploy/index.ts [hunk 2]
```diff
@@ -18,6 +18,11 @@ import { redeployCommand } from './command';
 import { getFlagsSpecification } from '../../util/get-flags-specification';
 import output from '../../output-manager';
 import { RedeployTelemetryClient } from '../../util/telemetry/commands/redeploy';
+import { CustomEnvironment } from '@vercel-internals/types';
```
Comment (erikareads) [inline packages/cli/src/commands/redeploy/index.ts]:
```suggestion
import type { CustomEnvironment } from '@vercel-internals/types';
```

**Review by erikareads — DISMISSED** [review]

**Review by erikareads — DISMISSED** [review]

File: .changeset/short-bananas-share.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'vercel': minor
+'@vercel/error-utils': patch
```
Comment (trek) [inline .changeset/short-bananas-share.md]: I don't see any changes to @vercel/error-utils. I don't think we need to bump version here.

**Review by trek — CHANGES_REQUESTED** [review]

File: .changeset/short-bananas-share.md (hunk 3 above)
Comment (paulogdm) [reply .changeset/short-bananas-share.md]: Yeah, not sure why this is there. Weird.

Comment (paulogdm) [inline .changeset/short-bananas-share.md]:
```suggestion
```

**Review by erikareads — APPROVED** [review]

**Review by trek — APPROVED** [review]

**Pull request merged**